src/main/java/pl/psnc/dei/ui/components/batches/ComplexBatchImportComponent.java

```diff
@@ -0,0 +1,170 @@
+        this.projectSelect = CommonComponentsFactory.getProjectSelector(this.projectsRepository);
+        this.projectSelect.setEmptySelectionAllowed(false);
+        this.projectSelect.addValueChangeListener(event -> {
+            Project project = this.projectSelect.getValue();
+            this.datasetSelect.setItems(this.datasetsRepository.findAllByProject(project));
+        });
+    }
+
+    private void prepareDatasetSelect() {
+        Project project = this.projectSelect.getValue();
+        this.datasetSelect =
+                CommonComponentsFactory.getDatasetSelector(
+                        (List<Dataset>) this.datasetsRepository.findAllByProject(project)
+                );
+    }
+
+    private void initUploads() {
+        this.prepareNameTextFiled();
+        this.prepareFileSelectButton();
+        this.prepareImportFile();
+        this.prepareImportButton();
+    }
+
+    private void prepareFileSelectButton() {
+        this.fileSelectButton = new Button();
+        this.fileSelectButton.setText("Upload");
+        this.fileSelectButton.setIcon(new Icon(VaadinIcon.CLOUD_UPLOAD_O));
+    }
+
+    private void prepareImportFile() {
+        this.memoryBuffer = new MemoryBuffer();
+        this.file = new Upload(this.memoryBuffer);
+        this.file.setMaxFiles(1);
+        this.file.setAcceptedFileTypes(".csv");
+        this.file.setDropAllowed(true);
+        this.file.setUploadButton(this.fileSelectButton);
+        this.file.addFileRejectedListener(
+                event -> Notification.show("File rejected. Reason: " + event.getErrorMessage(), 3000, Notification.Position.MIDDLE)
+        );
+    }
+
+    private void prepareNameTextFiled() {
+        this.nameTextFiled = new TextField();
+        this.nameTextFiled.setLabel("Import name");
+        this.nameTextFiled.addClassName("flex-1");
+        this.nameTextFiled.setMinLength(1);
+    }
+
+    private void prepareImportButton() {
+        this.importButton = new Button();
+        this.importButton.setText("Import");
+        this.importButton.addClickListener(
+                event -> {
+                    try {
+                        if (this.validate()) {
+                            this.batchService
+                                    .makeComplexImport(
+                                            this.memoryBuffer.getInputStream(),
+                                            this.nameTextFiled.getValue(),
+                                            this.projectSelect.getValue().getName(),
+                                            this.datasetSelect.getValue().getName()
+                                    );
+                            Notification.show("Import Finished!", 3000, Notification.Position.MIDDLE);
+                        }
+                        else Notification.show("Missing or wrong input", 3000, Notification.Position.MIDDLE);
+                    } catch (IOException e) {
+                        e.printStackTrace();
```

Comment:
In case of any exception during import processing, there is just stacktrace in logs. You should let user know, what happend when processing was interruped.

Reply:
more descriptive error messages added as notifications